sdk/src/metrics/state/observable_registry.cc

```diff
@@ -50,6 +50,7 @@ void ObservableRegistry::CleanupCallback(opentelemetry::metrics::ObservableInstr
 
 void ObservableRegistry::Observe(opentelemetry::common::SystemTimestamp collection_ts)
 {
+  static DefaultAttributesProcessor default_attribute_processor;
```

Comment:
What would happen without the default_attribute_processor specified here?

Reply:
They will get dropped by the [FilteredOrderedAttributeMap](https://github.com/open-telemetry/opentelemetry-cpp/blob/84e38b215f7b166e80353ff2c60836d9d375861d/sdk/src/metrics/state/filtered_ordered_attribute_map.cc#L19) which was added in earlier PR.